From: Stuart Marks <stuart.whitfield.marks@gmail.com>
Date: Sat, 5 Sep 2026 19:28:00 -0400
Subject: [PATCH] demux_mkv: fix compile with FFmpeg 8.0 by using AV_PROFILE_*

FFmpeg 8.0 removed the FF_PROFILE_ARIB_* macros and now requires using AV_PROFILE_ARIB_*.

diff --git a/demux/demux_mkv.c b/demux/demux_mkv.c
index d68ccf7c6c..227b79a203 100644
--- a/demux/demux_mkv.c
+++ b/demux/demux_mkv.c
@@ -2101,16 +2101,16 @@ static int demux_mkv_open_sub(demuxer_t *demuxer, mkv_track_t *track)
             // [0x30..0x37] are component tags utilized for
             // non-mobile captioning service ("profile A").
             if (component_tag >= 0x30 && component_tag <= 0x37)
-                lav->profile = FF_PROFILE_ARIB_PROFILE_A;
+                lav->profile = AV_PROFILE_ARIB_PROFILE_A;
             break;
         case 0x0012:
             // component tag 0x87 signifies a mobile/partial reception
             // (1seg) captioning service ("profile C").
             if (component_tag == 0x87)
-                lav->profile = FF_PROFILE_ARIB_PROFILE_C;
+                lav->profile = AV_PROFILE_ARIB_PROFILE_C;
             break;
         }
-        if (lav->profile == FF_PROFILE_UNKNOWN)
+        if (lav->profile == AV_PROFILE_UNKNOWN)
             MP_WARN(demuxer, "ARIB caption profile %02x / %04x not supported.\n",
                     component_tag, data_component_id);
     }

